### PR TITLE
Move split_utils.cuh to strings/detail

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Moving this header from `src/strings` to `include/cudf/strings/detail` to be accessed by the strings-udf work.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cudf/strings/detail/split_utils.cuh [hunk 1]
```diff
@@ -13,16 +13,19 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#pragma once
```
Comment (ttnghia) [inline cpp/include/cudf/strings/detail/split_utils.cuh:16]:
```suggestion

#pragma once
```

**Review by ttnghia — APPROVED** [review]

File: cpp/include/cudf/strings/detail/split_utils.cuh (hunk 1 above)
Comment (bdice) [reply cpp/include/cudf/strings/detail/split_utils.cuh:16]: We're highly inconsistent about whether a newline is needed after the license. I'd say this is not something worth worrying about in the future.
```bash
~/code/cudf $ rg "\*/\n#pragma once" --multiline --count | wc -l
186
~/code/cudf $ rg "\n\n#pragma once" --multiline --count | wc -l
240
```

**Review by bdice — CHANGES_REQUESTED** [review]
I have a somewhat-blocking request on the declaration of `position_pair`. Curious to hear your thoughts.

File: cpp/include/cudf/strings/detail/split_utils.cuh [hunk 2]
```diff
@@ -13,16 +13,19 @@
...
 namespace cudf {
 namespace strings {
 namespace detail {
 
-using position_pair = thrust::pair<size_type, size_type>;
+// JIT has trouble including thrust/pair.h
+using position_pair = struct {
+  size_type first;
+  size_type second;
+};
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/include/cudf/strings/detail/split_utils.cuh:28]: Yes, I like that. I'll make this change.

**Review by bdice — APPROVED** [review]

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**